### PR TITLE
Add tests for process header and extract nested values

### PR DESCRIPTION
**Pull request by AdrianMontaguSmartDCSIT**
This PR adds tests to the following functions to handle processing headers within json data converter:

- process_header
- extract_nested_value

### CODE REVIEW

File: tests/tel2puml/find_unique_graphs/test_json_data_converter.py [hunk 1]
```diff
@@ -222,3 +228,73 @@ def test_build_nested_dict() -> None:
...
+        # Test nested dictionary before and after list
+        path = "resource:attributes::service:name"
+        assert _extract_nested_value(sample_data, path) == {
+            "service:name": "frontend"
+        }
+
+        # Test larger nested data structure after list
+        path = "resource:attributes::spans"
+        assert _extract_nested_value(sample_data, path) == {
+            "spans": [{"trace_id": "001"}, {"span_id": "002"}]
```
Comment (FreddieMatherSmartDCSIT) [inline tests/tel2puml/find_unique_graphs/test_json_data_converter.py:267]: More of a question - why are we not keeping any of the paths in the return dict that come before a list?

**Review by FreddieMatherSmartDCSIT — COMMENTED** [review]
Just a question on the output dictionary key of extract nested value

File: tests/tel2puml/find_unique_graphs/test_json_data_converter.py (hunk 1 above)
Comment (AdrianMontaguSmartDCSIT) [reply tests/tel2puml/find_unique_graphs/test_json_data_converter.py:267]: The path before the list is the key within header_dict. In the function process_header we have:
```
    key = path.split("::")[0]
    if isinstance(value, (dict, list)):
        header_dict[key] = _flatten_json_dict(value)
    else:
        header_dict[key] = value
```
key is equal to the path before the list

Comment (FreddieMatherSmartDCSIT) [reply tests/tel2puml/find_unique_graphs/test_json_data_converter.py:267]: Ok think I understand

**Review by FreddieMatherSmartDCSIT — APPROVED** [review]
looks good to me

**Pull request merged**